datafusion/functions-nested/src/map_extract.rs

```diff
@@ -0,0 +1,170 @@
+    map_extract_udf
+);
+
+#[derive(Debug)]
+pub(super) struct MapExtract {
+    signature: Signature,
+    aliases: Vec<String>,
+}
+
+impl MapExtract {
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::user_defined(Volatility::Immutable),
+            aliases: vec![String::from("element_at")],
+        }
+    }
+}
+
+impl ScalarUDFImpl for MapExtract {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+    fn name(&self) -> &str {
+        "map_extract"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
+        let map_type = &arg_types[0];
+        let map_fields = get_map_entry_field(map_type)?;
+        Ok(map_fields[1].data_type().clone())
+    }
+
+    fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
+        make_scalar_function(map_extract_inner)(args)
+    }
+
+    fn aliases(&self) -> &[String] {
+        &self.aliases
+    }
+
+    fn coerce_types(&self, arg_types: &[DataType]) -> Result<Vec<DataType>> {
+        if arg_types.len() != 2 {
+            return exec_err!("map_extract expects two arguments");
+        }
+
+        let field = get_map_entry_field(&arg_types[0])?;
+        Ok(vec![
+            arg_types[0].clone(),
+            field.first().unwrap().data_type().clone(),
+        ])
+    }
+}
+
+fn map_extract_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
+    if args.len() != 2 {
+        return exec_err!("map_extract expects two arguments");
+    }
+
+    let map_array = as_map_array(&args[0])?;
+
+    if map_array.key_type() != args[1].data_type() {
+        return exec_err!(
+            "The key type {} does not match the map key type {}",
+            args[1].data_type(),
+            map_array.key_type()
+        );
+    }
+
+    let key_type = map_array.key_type();
+    if key_type.is_integer() {
+        generic_map_extract_inner::<Int64Type>(
+            map_array,
+            as_primitive_array::<Int64Type>(map_array.keys())?,
+            as_primitive_array::<Int64Type>(&args[1])?,
+        )
+    } else if key_type.is_floating() {
+        generic_map_extract_inner::<Float64Type>(
+            map_array,
+            as_primitive_array::<Float64Type>(map_array.keys())?,
+            as_primitive_array::<Float64Type>(&args[1])?,
+        )
+    } else if key_type.is_unsigned_integer() {
+        generic_map_extract_inner::<UInt32Type>(
+            map_array,
+            as_primitive_array::<UInt32Type>(map_array.keys())?,
+            as_primitive_array::<UInt32Type>(&args[1])?,
+        )
+    } else if key_type == &DataType::Utf8 {
+        string_map_extract_inner(
+            map_array,
+            as_string_array(map_array.keys())?,
+            as_string_array(&args[1])?,
+        )
+    } else {
+        exec_err!("Unsupported key type: {:?}", args[1].data_type())
+    }
```

Comment:
Can we use match statement here ? If will make more concise.
Also how about adding spport for `StringView` here